plugins/node/opentelemetry-instrumentation-pg/src/instrumentation.ts

```diff
@@ -173,101 +173,136 @@ export class PgInstrumentation extends InstrumentationBase {
-            plugin.getConfig(),
-            original,
-            ...args
-          );
+        // Modify query text w/ a tracing comment before invoking original for
+        // tracing, but only if args[0] has one of our expected shapes.
+        //
+        // TODO: remove the `as ...` casts below when the TS version is upgraded.
+        // Newer TS versions will use the result of firstArgIsQueryObjectWithText
+        // to properly narrow arg0, but TS 4.3.5 does not.
+        if (instrumentationConfig.addSqlCommenterCommentToQueries) {
+          args[0] = firstArgIsString
+            ? utils.addSqlCommenterComment(span, arg0 as string)
+            : firstArgIsQueryObjectWithText
+            ? {
+                ...(arg0 as utils.ObjectWithText),
+                text: utils.addSqlCommenterComment(
+                  span,
+                  (arg0 as utils.ObjectWithText).text
+                ),
+              }
+            : args[0];
         }
 
-        // Bind callback to parent span
+        // Bind callback (if any) to parent span (if any)
         if (args.length > 0) {
           const parentSpan = trace.getSpan(context.active());
           if (typeof args[args.length - 1] === 'function') {
             // Patch ParameterQuery callback
             args[args.length - 1] = utils.patchCallback(
-              plugin.getConfig(),
+              instrumentationConfig,
               span,
-              args[args.length - 1] as PostgresCallback
+              args[args.length - 1] as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span exists, bind the callback
             if (parentSpan) {
               args[args.length - 1] = context.bind(
                 context.active(),
                 args[args.length - 1]
               );
             }
-          } else if (
-            typeof (args[0] as NormalizedQueryConfig).callback === 'function'
-          ) {
+          } else if (typeof queryConfig?.callback === 'function') {
             // Patch ConfigQuery callback
             let callback = utils.patchCallback(
               plugin.getConfig(),
               span,
-              (args[0] as NormalizedQueryConfig).callback!
+              queryConfig.callback as PostgresCallback // nb: not type safe.
             );
+
             // If a parent span existed, bind the callback
             if (parentSpan) {
               callback = context.bind(context.active(), callback);
             }
 
-            // Copy the callback instead of writing to args.callback so that we don't modify user's
-            // original callback reference
+            // Copy the callback instead of writing to args.callback so that we
+            // don't modify user's original callback reference
             args[0] = { ...(args[0] as object), callback };
           }
         }
 
-        // Perform the original query
-        const result: unknown = original.apply(this, args as any);
+        if (
```

Comment:
need to add unittests for this hook. both for successful invocation and for the case where it throws. 
You can follow the tests for the `responseHook` instrumentation config option